src/main/java/com/atlassian/db/replica/internal/state/ConnectionState.java

```diff
@@ -224,32 +215,18 @@ private Connection prepareReadConnection(RouteDecisionBuilder decisionBuilder) t
             decisionBuilder.cause(writeConnection.getFirstCause().build());
             return writeConnection.get(decisionBuilder);
         }
-        final Optional<Database> consistentReplica;
+        boolean isConsistent;
         try {
-            final Collection<Database> replicas = cluster != null ?
-                cluster.getReplicas() :
-                Collections.singleton(new Database() {
-                    @Override
-                    public String getId() {
-                        return null;
-                    }
-
-                    @Override
-                    public Supplier<Connection> getConnectionSupplier() {
-                        return () -> readConnection.get(decisionBuilder);
-                    }
-                });
-            consistentReplica = consistency.getConsistent(replicas);
+            isConsistent = consistency.isConsistent(() -> readConnection.get(decisionBuilder));
         } catch (Exception e) {
             closeConnection(readConnection, decisionBuilder);
             throw e;
         }
-        if (consistentReplica.isPresent()) {
+        if (isConsistent) {
```

Comment:
This is a bit long. Perhaps we could extract the if to a private method?

Reply:
It' just reverts to the previous state. It can be refactored on a separate PR.